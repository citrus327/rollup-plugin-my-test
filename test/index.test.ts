import { describe, it, afterEach, expect } from "vitest";
import { rollup, RollupBuild, OutputOptions, RollupOptions } from "rollup";
import commonjs from "@rollup/plugin-commonjs";
import nodeResolve from "@rollup/plugin-node-resolve";
import myPlugin from "../src/index";
import path from "path";

const inputOptions: RollupOptions = {
  input: "test/manifest/index.js",
  plugins: [
    commonjs(),
    nodeResolve(),
    myPlugin({
      entries: [
        { find: "@test", replacement: path.resolve(process.cwd(), "./test") },
      ],
    }),
  ],
};

const outputOptions: OutputOptions[] = [
  {
    file: "test/dist/bundle.cjs",
    format: "cjs",
  },
  {
    file: "test/dist/bundle.mjs",
    format: "es",
  },
];

let bundle: RollupBuild;

describe("suite", () => {
  afterEach(() => {
    bundle?.close();
  });

  it("should passes", async () => {
    bundle = await rollup(inputOptions);
    const result = await Promise.all(
      outputOptions.map((opt) => {
        bundle.write(opt);
        return bundle.generate(opt);
      })
    );
    expect(result).toBeTruthy();
  });
});
