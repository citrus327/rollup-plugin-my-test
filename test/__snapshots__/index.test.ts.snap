// Vitest Snapshot v1

exports[`suite > should passes 1`] = `
[
  {
    "output": [
      {
        "code": "'use strict';

var foo = \\"hello world!\\";

const a = () => {
  console.log(\`i'm function A\`);
};

var a_1 = a;

function test() {
  console.log(foo);
}

exports.a = a_1;
exports.test = test;
",
        "dynamicImports": [],
        "exports": [
          "a",
          "test",
        ],
        "facadeModuleId": "/Users/hao/spaces/projj/github.com/citrus327/rollup-plugin-my-test/test/src/index.js",
        "fileName": "bundle.cjs",
        "implicitlyLoadedBefore": [],
        "importedBindings": {},
        "imports": [],
        "isDynamicEntry": false,
        "isEntry": true,
        "isImplicitEntry": false,
        "map": null,
        "moduleIds": [
          "/Users/hao/spaces/projj/github.com/citrus327/rollup-plugin-my-test/test/src/foo.js",
          "/Users/hao/spaces/projj/github.com/citrus327/rollup-plugin-my-test/node_modules/.pnpm/a-treeshakable-pkg@1.0.0/node_modules/a-treeshakable-pkg/dist/bundle.js",
          "/Users/hao/spaces/projj/github.com/citrus327/rollup-plugin-my-test/test/src/index.js",
        ],
        "modules": {
          "/Users/hao/spaces/projj/github.com/citrus327/rollup-plugin-my-test/node_modules/.pnpm/a-treeshakable-pkg@1.0.0/node_modules/a-treeshakable-pkg/dist/bundle.js": {
            "code": "const a = () => {
  console.log(\`i'm function A\`);
};

var a_1 = a;",
            "originalLength": 225,
            "removedExports": [
              "__moduleExports",
              "b",
              "c",
              "default",
            ],
            "renderedExports": [
              "a",
            ],
            "renderedLength": 67,
          },
          "/Users/hao/spaces/projj/github.com/citrus327/rollup-plugin-my-test/test/src/foo.js": {
            "code": "var foo = \\"hello world!\\";",
            "originalLength": 31,
            "removedExports": [],
            "renderedExports": [
              "default",
            ],
            "renderedLength": 25,
          },
          "/Users/hao/spaces/projj/github.com/citrus327/rollup-plugin-my-test/test/src/index.js": {
            "code": "function test() {
  console.log(foo);
}",
            "originalLength": 131,
            "removedExports": [],
            "renderedExports": [
              "test",
              "a",
            ],
            "renderedLength": 39,
          },
        },
        "name": "index",
        "referencedFiles": [],
        "type": "chunk",
      },
    ],
  },
  {
    "output": [
      {
        "code": "var foo = \\"hello world!\\";

const a = () => {
  console.log(\`i'm function A\`);
};

var a_1 = a;

function test() {
  console.log(foo);
}

export { a_1 as a, test };
",
        "dynamicImports": [],
        "exports": [
          "a",
          "test",
        ],
        "facadeModuleId": "/Users/hao/spaces/projj/github.com/citrus327/rollup-plugin-my-test/test/src/index.js",
        "fileName": "bundle.mjs",
        "implicitlyLoadedBefore": [],
        "importedBindings": {},
        "imports": [],
        "isDynamicEntry": false,
        "isEntry": true,
        "isImplicitEntry": false,
        "map": null,
        "moduleIds": [
          "/Users/hao/spaces/projj/github.com/citrus327/rollup-plugin-my-test/test/src/foo.js",
          "/Users/hao/spaces/projj/github.com/citrus327/rollup-plugin-my-test/node_modules/.pnpm/a-treeshakable-pkg@1.0.0/node_modules/a-treeshakable-pkg/dist/bundle.js",
          "/Users/hao/spaces/projj/github.com/citrus327/rollup-plugin-my-test/test/src/index.js",
        ],
        "modules": {
          "/Users/hao/spaces/projj/github.com/citrus327/rollup-plugin-my-test/node_modules/.pnpm/a-treeshakable-pkg@1.0.0/node_modules/a-treeshakable-pkg/dist/bundle.js": {
            "code": "const a = () => {
  console.log(\`i'm function A\`);
};

var a_1 = a;",
            "originalLength": 225,
            "removedExports": [
              "__moduleExports",
              "b",
              "c",
              "default",
            ],
            "renderedExports": [
              "a",
            ],
            "renderedLength": 67,
          },
          "/Users/hao/spaces/projj/github.com/citrus327/rollup-plugin-my-test/test/src/foo.js": {
            "code": "var foo = \\"hello world!\\";",
            "originalLength": 31,
            "removedExports": [],
            "renderedExports": [
              "default",
            ],
            "renderedLength": 25,
          },
          "/Users/hao/spaces/projj/github.com/citrus327/rollup-plugin-my-test/test/src/index.js": {
            "code": "function test() {
  console.log(foo);
}",
            "originalLength": 131,
            "removedExports": [],
            "renderedExports": [
              "test",
              "a",
            ],
            "renderedLength": 39,
          },
        },
        "name": "index",
        "referencedFiles": [],
        "type": "chunk",
      },
    ],
  },
]
`;
